HousingFinanceInterimApi/V1/UseCase/MoveHousingBenefitFileUseCase.cs

```diff
@@ -172,10 +176,22 @@ private async Task<List<GoogleFileSettingDomain>> GetGoogleFileSetting(string la
 
         private string CalculateNewFileName(File file)
         {
-            var fileNameDateMatches = _academyFilePattern.Matches(file.Name);
-            var fileNameDate = DateTime.ParseExact(fileNameDateMatches[1].Value, "ddMMyyyy", null);
+            // Handle null creation date
+            if (file.CreatedTime == null)
+            {
+                Console.WriteLine($"File {file.Name} has no creation date");
+                return null;
```

Comment:
I'm a bit worried about this `null` part.

I wonder if it will trip up this line (Ln117) and the logic further: 
https://github.com/LBHackney-IT/housing-finance-interim-api/blob/feac200ee3f439ffcf6a3d5159d5d4844fdd6279/HousingFinanceInterimApi/V1/UseCase/MoveHousingBenefitFileUseCase.cs#L117-L119
So, is the Regex still valid? If not, will it throw? If it's valid & it will not throw, how will it affect match checks after it?

Given, the behaviour of this helper function has drastically changed, I think the overall UC needs a test to cover this case. _(Pretty much, we went from the function always returning a value, to it sometimes returning no value.)_

A thing worth considering is: what do we want the UC to do, when file has no 'CreatedTime'? For example, if the 'CreatedTime' is `null`, do we want that file to be ignored? Do we want to throw an error? Or do we want the logic to fallback on the previous, now deleted logic? This:
``` cs
return $"HousingBenefitFile{fileNameDate.AddDays(-7).ToString("yyyyMMdd")}.dat";
```

Reply:
I think it's impossible for a file to not have a created time, but the IDE treats it as nullable. Do you know how else I could get around this?

Reply:
Just left another comment about it :D You should check the `File` class, see what kind of field it is. Is it `string`, `DateTime` or `DateTime?`?

Reply:
The thing is... if the `File` class belongs to the `Google` namespace, then it was set by whoever is in charge of representing the GDrive. If they made it nullable, then it's likely there some cases where it can be `null`. I suppose, we won't get to learn whether our case avoids the `null` possibility, but at the very least, we can put in a defensive code clause to protect from it. Though, this requires consideration of: what we want to do with the file, when its new file name cannot be calculated?

Reply:
` /// <summary>The time at which the file was created (RFC 3339 date-time).</summary>
        [Newtonsoft.Json.JsonPropertyAttribute("createdTime")]
        public virtual string CreatedTimeRaw { get; set; }

        /// <summary><seealso cref="System.DateTime"/> representation of <see cref="CreatedTimeRaw"/>.</summary>
        [Newtonsoft.Json.JsonIgnoreAttribute]
        public virtual System.Nullable<System.DateTime> CreatedTime
        {
            get => Google.Apis.Util.Utilities.GetDateTimeFromString(CreatedTimeRaw);
            set => CreatedTimeRaw = Google.Apis.Util.Utilities.GetStringFromDateTime(value);
        }`
It is Google's file class. Based on this it can theoretically be null but it shouldn't ever be the case for this usecase, because we're pulling the file directly from Google drive.

Reply:
My thinking is that the exception probably makes the most sense. I mean:
`No CreatedTime --> No Monday Calculation --> No 'New file name' --> No Name to Copy file with --> If name doesn't match the typical new name format, then the processing of the file won't get triggered anyway`

Reply:
Great! Would be good to write a test that covers this scenario though.

Reply:
Added a test for it. `UCThrowsExceptionWhenAcademyFileCreationTimeIsNull`